### PR TITLE
test: book-db-integration

### PR DESCRIPTION
**Pull request by Urhengulas**
* `book`: Add `mod test_data` (hidden behind feature "test-data")
* `book::tests::end_to_end`: Setup end-to-end testing
    * 2 tests:
        * `should_not_find_book`
        * `should_find_book`
    * `utils`
        * Start rpc server and connect client
        * Create dummy database + run migrations
        * fill test data

### CODE REVIEW

File: book/Cargo.toml [hunk 1]
```diff
@@ -24,3 +24,9 @@ serde = { version = "1.0", features = ["derive"] }
 tarpc = { version = "0.22", features = ["full"] }
 tokio = { version = "0.2", features = ["full"] }
 tokio-serde = { version = "0.6", features = ["json"] }
+
+[dev-dependencies]
+uuid = { version = "0.8", features = ["v4"] }
+
+[features]
+test-data = []
```
Comment (code-mm) [inline book/Cargo.toml:32]: The PR description says that the feature is named `testing`.

File: book/src/lib/lib.rs [hunk 2]
```diff
@@ -1,5 +1,7 @@
 pub mod db;
 mod rpc;
+#[cfg(feature = "test-data")]
```
Comment (code-mm) [inline book/src/lib/lib.rs:3]: Feature name?

**Review by code-mm — CHANGES_REQUESTED** [review]

File: book/src/lib/lib.rs (hunk 2 above)
Comment (Urhengulas) [reply book/src/lib/lib.rs:3]: the "feature" is fixed syntax. I can't rename that.

File: book/Cargo.toml (hunk 1 above)
Comment (Urhengulas) [reply book/Cargo.toml:32]: Will update the description

File: book/src/lib/lib.rs (hunk 2 above)
Comment (code-mm) [reply book/src/lib/lib.rs:3]: I know that, this was related to the `test-data`.

**Review by code-mm — APPROVED** [review]

**Pull request merged**